nixos/modules/programs/ghostty.nix

```diff
@@ -0,0 +1,47 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.ghostty;
+in
+{
+  options.programs.ghostty = {
+    enable = lib.mkEnableOption "ghostty terminal emulator";
+
+    package = lib.mkPackageOption pkgs "ghostty" { };
+
+    settings = lib.mkOption {
+      type =
+        with lib.types;
+        oneOf [
+          singleLineStr
+          (listOf singleLineStr)
+        ];
```

Comment:
There should be a types.attrsOf here, no? (Also types.oneOf with two branches = types.either)